pyproject.toml

```diff
@@ -121,6 +127,113 @@ docs = [
 ]
 codegen = ["rich", "pydantic", "jinja2>=3.1.6"]
 
+[project.optional-dependencies]
+ui = [
+    "streamlit",
+    "pandas",
+    "llama-stack-client>=0.2.8",
+    "streamlit-option-menu",
+]
+
+# Provider-specific optional dependencies
+# These can be used for common provider combinations
+# To install the "ollama" distro just run:
+#  uv venv --python 3.10 ollama-distro
+#  source ollama-distro/bin/activate
+#  uv sync --active --extra ollama
+
+#######################
+# INFERENCE PROVIDERS #
+#######################
+inference-remote-ollama = ["ollama", "aiohttp", "h11>=0.16.0"]
+
+#######################
+# VECTOR IO PROVIDERS #
+#######################
+vector-io-inline-faiss = ["faiss-cpu"]
+vector-io-remote-chroma = ["chromadb-client"]
+vector-io-remote-pgvector = ["psycopg2-binary"]
+
+####################
+# AGENTS PROVIDERS #
+####################
+agents-inline-meta-reference = [
+    "matplotlib",
+    "pillow",
+    "pandas",
+    "scikit-learn",
+]
+
+#######################
+# TELEMETRY PROVIDERS #
+#######################
+telemetry-inline-opentelemetry = [
+    "opentelemetry-sdk",
+    "opentelemetry-exporter-otlp-proto-http",
+]
+
+########################
+# EVALUATION PROVIDERS #
+########################
+eval-inline-meta-reference = [
+    "tree_sitter",
+    "pythainlp",
+    "langdetect",
+    "emoji",
+    "nltk",
+]
+
+#######################
+# DATASETIO PROVIDERS #
+#######################
+datasetio-inline-local-fs = ["pandas"]
+datasetio-remote-huggingface = ["datasets"]
+
+#####################
+# SCORING PROVIDERS #
+#####################
+scoring-inline-braintrust = ["autoevals", "openai"]
+
+###########################
+# POST-TRAINING PROVIDERS #
+###########################
+post-training-inline-huggingface = [
+    "torch",
+    "trl",
+    "transformers",
+    "peft",
+    "datasets",
+]
+
+##########################
+# TOOL RUNTIME PROVIDERS #
+##########################
+tool-runtime-inline-bing-search = ["requests"]
+tool-runtime-inline-brave-search = ["requests"]
+tool-runtime-inline-tavily-search = ["requests"]
+tool-runtime-inline-wolfram-alpha = ["requests"]
+tool-runtime-inline-model-context-protocol = ["mcp"]
+tool-runtime-inline-rag-runtime = [
+    "blobfile",
+    "chardet",
+    "pypdf",
+    "tqdm",
+    "numpy",
+    "scikit-learn",
+    "scipy",
+    "nltk",
+    "sentencepiece",
+    "transformers",
+]
+
+####################################
+# COMMON DISTRIBUTION COMBINATIONS #
+####################################
+# All declarations must be a single line to comply with pep508...
+ollama = [
```

Comment:
so I think we could produce this automatically as an edit to pyproject.toml -- whatever the `dependencies.json` was trying to do for all our distros?